Cython/Compiler/PyrexTypes.py

```diff
@@ -695,7 +695,6 @@ def is_complete(self):
     def declaration_code(self, entity_code,
             for_display = 0, dll_linkage = None, pyrex = 0):
         # XXX: we put these guards in for now...
-        assert not pyrex
```

Comment:
I guess the call in line 3229 of the same file is the culprit here. There is a special-case for `for_display` just below. The question is how `pyrex=True` should behave there (read: the declaration code in a Cython-specific context).

What happens if you create multiple wrappers for functions with slightly different memoryview arguments? Do we build suitable C names for all of them?

Reply:
The call in line 3229 is the culprit.

Multiple wrappers for functions create multiple wrappers with the same name that all conflict with each other. Adding `for_display = for_display or pyrex` creates something that compiles (but probably doesn't really do what `pyrex` implies it should). I'll come back to this in the individual PR.

Reply:
`pyrex=True` is usually very similar to `for_display=True`, since both target the Cython code level. See the comment near the top of the file in `class PyrexType`. Can't say right now what a good escaped version of a memoryview declarations for use in a C name should look like. Probably needs some trial and error.